src/servers/graphql-server.ts

```diff
@@ -92,31 +92,42 @@ export const startApolloServer = async ({
         ip = ips
       }
 
-      let wallet, user
+      let wallet
 
       // TODO move from id: uuidv4() to a Jaeger standard
       const logger = graphqlLogger.child({ token, id: uuidv4(), body: context.req?.body })
 
       let domainUser: User | null = null
       return addAttributesToCurrentSpanAndPropagate(
-        { [SemanticAttributes.ENDUSER_ID]: uid, [SemanticAttributes.HTTP_CLIENT_IP]: ip },
+        {
+          [SemanticAttributes.ENDUSER_ID]: userId,
+          [SemanticAttributes.HTTP_CLIENT_IP]: ip,
+        },
         async () => {
-          if (uid) {
-            const loggedInUser = await Users.getUserForLogin({ userId: uid, ip })
-            if (loggedInUser instanceof Error)
+          if (userId) {
+            const lastConnection = new Date()
+
+            const user = await UsersRepository().findByIdAndUpdateLastConnectionDate(
+              userId,
+              lastConnection,
+            )
+
+            if (user instanceof Error)
               throw new ApolloError(
                 "Invalid user authentication",
                 "INVALID_AUTHENTICATION",
                 {
-                  reason: loggedInUser,
+                  reason: user,
                 },
               )
-            domainUser = loggedInUser
-            user = await User.findOne({ _id: uid })
+            domainUser = user
+
             wallet =
-              !!user && user.status === "active"
-                ? await WalletFactory({ user, logger })
+              !!domainUser && domainUser.status === "active"
+                ? await WalletFactory({ user: domainUser, logger })
```

Comment:
This is broken `domainUser` is not the correct type for the `WalletFactory` - getting rid of the `WalletFactory` is an ongoing cleanup that Arvin is working on.

Reply:
what is this breaking? v1?

Reply:
Any calls to the wallet factory that call the internal user reference. The object schema of `user` and `domainUser` are not identical.